service/controller/v5/resource/blobobject/blob_client.go

```diff
@@ -0,0 +1,119 @@
+package blobobject
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2018-07-01/storage"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+)
+
+const (
+	blobFormatString      = `https://%s.blob.core.windows.net`
+	resourceNotFoundError = "NotFound"
+	maxRetriesRequests    = 3
+)
+
+type BlobClient struct {
+	storageAccountsClient *storage.AccountsClient
+	containerURL          azblob.ContainerURL
+}
+
+func (c *BlobClient) BlobExists(ctx context.Context, blobName string) (bool, error) {
+	blobURL := c.containerURL.NewBlockBlobURL(blobName)
+
+	_, err := blobURL.GetProperties(ctx, azblob.BlobAccessConditions{})
+	if IsBlobNotFound(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, microerror.Mask(err)
+	}
+
+	return true, nil
+}
+
+func (c *BlobClient) createBlockBlob(ctx context.Context, blobName string, payload string) (azblob.BlockBlobURL, error) {
+	blob := c.containerURL.NewBlockBlobURL(blobName)
+
+	_, err := blob.Upload(
+		ctx,
+		strings.NewReader(payload),
+		azblob.BlobHTTPHeaders{
+			ContentType: "text/plain",
+		},
+		azblob.Metadata{},
+		azblob.BlobAccessConditions{},
+	)
+
+	return blob, err
+}
+
+func (c *BlobClient) getAccountPrimaryKey(ctx context.Context, accountName, accountGroupName string) (string, error) {
+	keys, err := c.storageAccountsClient.ListKeys(ctx, accountGroupName, accountName)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+	return *(((*keys.Keys)[0]).Value), nil
```

Comment:
That looks like we should have more error handling for the list length check and type assertion.

Reply:
API by spec returns two keys. There is no state you can delete key or so. So there are always two keys returned if no error

Reply:
I understand that. We try to have our code to be defensive though and deal with more weirdnesses than we anticipate. This makes the platform more reliable.